src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -4097,6 +4252,11 @@ public List<TElement> ToList()
             return list;
         }
 
+        public int Count
+        {
+            get { return source.Count(); }
+        }
```

Comment:
As in the other case above, usage of this IListProvider.Count seem to assume that Count is cheap, but it's not here.  Seems like you may need to make this a method like ```bool TryGetCount(out int count)``` or something like that.